packages/cashier/src/Components/Form/__tests__/payment-agent-transfer-form.spec.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import PaymentAgentTransferForm from '../payment-agent-transfer-form';
+
+jest.mock('Stores/connect', () => ({
+    __esModule: true,
+    default: 'mockedDefaultExport',
+    connect: () => Component => Component,
+}));
+
+jest.mock('@deriv/shared/src/utils/validation/declarative-validation-rules', () => ({
+    __esModule: true,
+    default: 'mockedDefaultExport',
+    validNumber: jest.fn(() => true),
+}));
+
+describe('<PaymentAgentTransferForm />', () => {
+    const balance = '80';
+    const setErrorMessage = jest.fn();
+    const transfer_limit = {
+        min: '1',
+        max: '100',
+    };
+
+    beforeAll(() => {
+        const modal_root_el = document.createElement('div');
+        modal_root_el.setAttribute('id', 'modal_root');
+        document.body.appendChild(modal_root_el);
+    });
+
+    afterAll(() => {
+        document.body.removeChild(modal_root_el);
+    });
+
+    it('should render the component', () => {
+        const { container } = render(<PaymentAgentTransferForm />);
+
+        expect(container.firstChild).toHaveClass('payment-agent-transfer-form__container');
+    });
+
+    it('should show an error if client login id or amount is not provided', async () => {
+        render(
+            <PaymentAgentTransferForm
+                balance={balance}
+                setErrorMessage={setErrorMessage}
+                transfer_limit={transfer_limit}
+            />
+        );
+
+        const submit_button = screen.getByRole('button');
+        fireEvent.click(submit_button);
+
+        await waitFor(() => {
+            expect(screen.getByText('Please enter a valid client login ID.')).toBeInTheDocument();
+            expect(screen.getByText('This field is required.')).toBeInTheDocument();
+        });
+    });
+
+    it('should show an error if the login id is not valid', async () => {
+        const { container } = render(
+            <PaymentAgentTransferForm
+                balance={balance}
+                setErrorMessage={setErrorMessage}
+                transfer_limit={transfer_limit}
+            />
+        );
+
+        const loginid_field = container.querySelector('input[name=loginid]');
+        const submit_button = screen.getByRole('button');
```

Comment:
@farrah-deriv Since this line of code is written 4 times, don't you think it's better to move it to the top?

Reply:
`screen` is dependent on the component that is rendered on every test so I can't declare it at the top

Reply:
hmmmm, ok then👍